### PR TITLE
fix(network): fall back to ethtool ioctl when sysfs perm_addr is missing

### PR DESCRIPTION
**Pull request by IvanHunters**
## Problem

`getPermanentMAC` reads `/sys/class/net/<if>/perm_addr` only. On some
driver/kernel combinations that file does not exist, or contains the
all-zero address. The current fallback returns the active MAC, but for a
bond slave the active MAC has already been rewritten to the bond master's
MAC. The result is that both slaves of an LACP bond resolve to the same
`enx<mac>` predictable name, and `GenerateBondCmdline` emits something like:

    bond=bond0:enxd283bcf788ae,enxd283bcf788ae:mode=802.3ad,...

After kexec into Talos, the bond cannot be assembled from a single
duplicated slave name, the VLAN never comes up, and the host loses network.

## Fix

Add an `ETHTOOL_GPERMADDR` ioctl fallback to `getPermanentMAC`. The ethtool
interface returns the burned-in hardware address directly from the driver
and is unaffected by bonding's MAC rewrite. The sysfs path remains the
primary source; ethtool is consulted only when sysfs is missing or returns
all zeros.

For non-Ethernet master devices (bond, VLAN, bridge) ethtool also returns
all zeros, which is caught and propagated as an error so the existing
`net.InterfaceByName` fallback in `PrettyName` keeps its current behaviour.

## Validation

- New `TestIsZeroMAC` covers the zero-detection helper.
- Existing `internal/network` tests pass on linux/amd64.
- Verified on an affected host (kernel 6.8, Intel igb, 802.3ad bond):
  before the fix, sysfs `perm_addr` is absent and both slaves report the
  bond master MAC via sysfs `address`; after the fix, the ioctl returns
  the correct per-slave permanent MACs and the generated cmdline lists two
  distinct `enx<mac>` names. End-to-end kexec into Talos confirmed the
  bond/VLAN/IP stack comes up correctly.

## Notes

- No new dependencies; uses `golang.org/x/sys/unix` already present.
- Linux-only file, matches the existing `//go:build linux` constraint.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved permanent MAC address detection by adding a more reliable lookup flow with stronger validation, ensuring the correct burned-in hardware address is used when available.

* **Tests**
  * Added unit test coverage for MAC address “zero/invalid” detection to better confirm correct handling of edge cases.

### CODE REVIEW

File: internal/network/network.go [hunk 1]
```diff
@@ -586,17 +587,100 @@ func IfaceAddr(name string) (ip, mask string, err error) {
...
+		if b != 0 {
+			return false
+		}
+	}
+	return true
+}
+
+// ethtoolIfreq mirrors struct ifreq with the ifr_data union pointer used by
+// SIOCETHTOOL. The Name field is null-terminated and bounded by IFNAMSIZ.
+type ethtoolIfreq struct {
```
Comment (lexfrei) [inline internal/network/network.go:640]: The kernel copies `sizeof(struct ifreq)` from user space on the SIOCETHTOOL path (`get_user_ifreq` → `copy_from_user(ifr, arg, sizeof(*ifr))`). On amd64 `struct ifreq` is 40 bytes (16-byte name + 24-byte `ifr_ifru` union), but this struct is only 24 bytes (16 + an 8-byte pointer), so the kernel reads 16 bytes past its end on every ethtool fallback call.

`golang.org/x/sys/unix` — already a dependency here — guards against exactly this. Its `ifreqData` pads to the full size:

```go
type ifreqData struct {
    name [IFNAMSIZ]byte
    data unsafe.Pointer
    _    [len(ifreq{}.Ifru) - SizeofPtr]byte // Pad to the same size as ifreq.
}
```

and `ztypes_linux_amd64.go` confirms `ifreq{ Ifrn [16]byte; Ifru [24]byte }` (40 bytes total).

In practice the over-read lands in adjacent mapped memory that SIOCETHTOOL ignores, so it works — and the live-node test does not exercise the failure mode. But it is an out-of-bounds read at the kernel boundary that can EFAULT if the struct ever sits at the tail of a mapped region.

Fix — pad to `sizeof(struct ifreq)`, mirroring `x/sys/unix`:

```go
type ethtoolIfreq struct {
    Name [unix.IFNAMSIZ]byte
    Data unsafe.Pointer
    _    [16]byte // pad to sizeof(struct ifreq) on 64-bit linux: 16 + 8 + 16 = 40
}
```

**Review by lexfrei — CHANGES_REQUESTED** [review]
NOT LGTM — the ethtool fallback passes an undersized `struct ifreq` to the kernel, causing a 16-byte out-of-bounds read on every ethtool call.

**Business context**: on kernels/drivers where sysfs `perm_addr` is missing or zero, both LACP bond slaves resolved to the same `enx<master_mac>` name, breaking the `bond=` cmdline and the network after kexec into Talos. The fix and its direction are correct and verified on a live node.

## Blockers

### B1: `ethtoolIfreq` is smaller than the kernel's `struct ifreq`

`internal/network/network.go:636` — see the inline comment for the mechanism, evidence, and fix.

## Non-blocking follow-ups

1. The inline suggestion to switch to `unix.IoctlSetPointer` would not compile: that function does not exist in `golang.org/x/sys/unix` (only `IoctlSetPointerInt`, which takes an `int`, not a pointer). The type-safe `ifreqData`/`withData` path is unexported, so there is no public helper for this ioctl, and the raw `unix.Syscall(SYS_IOCTL, …)` here is acceptable. The real issue is the struct size (B1), not the call mechanism.
2. This is internal MAC-detection plumbing and the `enx<mac>` naming format is unchanged, so no documentation update is required.

**Review by lexfrei — APPROVED** [review]
LGTM — the undersized `struct ifreq` flagged earlier is now padded to the full kernel struct size, derived portably from `unix.Ifreq`, and locked in by a size guard test. No blockers remain.

**Business context**: on kernels/drivers where sysfs `perm_addr` is missing or zero, both LACP bond slaves resolved to the same `enx<master_mac>` name, breaking the `bond=` cmdline and the network after kexec into Talos.

The previous blocker (`ethtoolIfreq` smaller than the kernel's `struct ifreq`, 16-byte out-of-bounds read on the SIOCETHTOOL path) is resolved: the trailing pad field makes `sizeof(ethtoolIfreq) == sizeof(struct ifreq)`, and `TestEthtoolIfreqSizeMatchesKernel` prevents regression.

Verified: vet/build/lint clean, the `internal/network` tests pass on linux/arm64, and the size guard cross-compiles cleanly on 386/arm/arm64/ppc64le/s390x/riscv64.
